polyphemus/lib/etls/redcap/lib/script.rb

```diff
@@ -1,37 +1,104 @@
 module Redcap
   class Script
-    def initialize(model, script, redcap_template)
+    def initialize(model, script, template)
       @model = model
-      @script = script
-      @redcap_template = redcap_template
+      @project = model.project
+      @template = template
+      @attributes ||= script.map do |att_name, att_value|
+        [ att_name, Redcap::Value.new(att_name, att_value, template) ]
+      end.to_h
     end
 
-    def forms
-      @forms ||= @script[:forms].map do |form_name, form|
-        [ form_name, Redcap::Form.new(@model, form_name, form, @redcap_template) ]
-      end.to_h
+    def fields
+      @fields ||= @attributes.values.map(&:field_name).uniq
+    end
+
+    def records
+      eavs = @project.eav_records.select do | eav|
+        fields.include?(eav[:field_name])
+      end
+
+      return {} unless eavs
+
+      records = eavs.group_by do |eav|
+        @model.events? ?  [ eav[:record], eav[:redcap_event_name] ] : eav[:record]
+      end.map do |record_id, record_eavs|
+        [
+          record_id,
+          Redcap::Client::Record.new(
+            record_eavs,
+            @project.flat_records[record_id]
+          ).record
+        ]
+      end.to_h.compact
+
+      return records
+    end
+
+    def inverse_load
+      update = {}
+
+      redcap_records = records
+
+      @model.magma_records.each do |record_name, magma_record|
+        next unless @attributes.values.all?{|v| v.valid?(magma_record) }
```

Comment:
I am a bit confused about on thing however-- the magma_records line seems to query only identifiers, will these magma_record entries support this attributes matching algorithm for other attributes somehow?

We definitely shouldn't query "all" attributes, but we may need, another approach as well?